raft/src/main/java/org/apache/kafka/raft/QuorumConfig.java

```diff
@@ -102,6 +103,11 @@ public class QuorumConfig {
     public static final String QUORUM_RETRY_BACKOFF_MS_DOC = CommonClientConfigs.RETRY_BACKOFF_MS_DOC;
     public static final int DEFAULT_QUORUM_RETRY_BACKOFF_MS = 20;
 
+    public static final String QUORUM_AUTO_JOIN_ENABLE_CONFIG = QUORUM_PREFIX + "auto.join.enable";
```

Comment:
@kevin-wu24 it would be cool if users could find out about this new config through upgrade.html. Could you please consider addressing it in the follow-up?